File: devscripts/create_release.pl
#! /usr/bin/env perl

use strict;

my $build_cmd = "python -m build";
my $twine_cmd = "python -m twine upload --repository testpypi dist/*";

my $err = system($build_cmd);
if ($err == 0) {
    system($twine_cmd);
}
else {
    die("$build_cmd failed. Not uploading to PyPi");
}

File: devscripts/create_release.py
#!/usr/bin/env python

import argparse
import os
import subprocess

parser = argparse.ArgumentParser()
parser.add_argument('--production', action='store_true')
parser.add_argument('--clean', action='store_true')
args = parser.parse_args()

if args.clean:
    if os.path.isdir('./dist'):
        subprocess.run(['rm', 'dist/*'], check=True, shell=True)

build_cmd = 'python -m build'
testpypi_twine_cmd = 'python -m twine upload --repository testpypi dist/*'
production_twine_cmd = 'python -m twine upload dist/*'
twine_cmd = testpypi_twine_cmd if not args.production else production_twine_cmd

try:
    subprocess.run(build_cmd, check=True, shell=True)
    subprocess.run(twine_cmd, check=True, shell=True)
except subprocess.CalledProcessError:
    raise SystemExit(f'{build_cmd} failed. Not uploading to PyPi')

